src/main/java/ch/njol/yggdrasil/Fields.java

```diff
@@ -182,130 +178,132 @@ public Fields() {
+	 * @param object Some object
+	 * @throws NotSerializableException If a field occurs more than once (i.e. if a class
+	 *                                  has a field with the same name as a field in one of its superclasses)
 	 */
-	public Fields(final Object o, @Nullable final Yggdrasil yggdrasil) throws NotSerializableException {
+	public Fields(Object object, @Nullable Yggdrasil yggdrasil) throws NotSerializableException {
 		this.yggdrasil = yggdrasil;
-		final Class<?> c = o.getClass();
-		assert c != null;
-		for (final Field f : getFields(c)) {
-			assert f != null;
+		Class<?> type = object.getClass();
+		assert type != null;
+		for (Field field : getFields(type)) {
+			assert field != null;
 			try {
-				fields.put(Yggdrasil.getID(f), new FieldContext(f, o));
-			} catch (final IllegalArgumentException e) {
-				assert false;
-			} catch (final IllegalAccessException e) {
+				fields.put(Yggdrasil.getID(field), new FieldContext(field, object));
+			} catch (IllegalArgumentException | IllegalAccessException e) {
 				assert false;
 			}
 		}
 	}
 	
-	private final static Map<Class<?>, Collection<Field>> cache = new HashMap<>();
+	private static final Map<Class<?>, Collection<Field>> cache = new HashMap<>();
 	
 	/**
-	 * Gets all serialisable fields of the provided class, including superclasses.
+	 * Gets all serializable fields of the provided class, including superclasses.
 	 * 
-	 * @param c The class to get the fields of
+	 * @param type The class to get the fields of
 	 * @return All non-static and non-transient fields of the given class and its superclasses
-	 * @throws NotSerializableException If a field occurs more than once (i.e. if a class has a field with the same name as a field in one of its superclasses)
+	 * @throws NotSerializableException If a field occurs more than once (i.e. if a class has a
+	 *                                  field with the same name as a field in one of its superclasses)
 	 */
-	public static Collection<Field> getFields(final Class<?> c) throws NotSerializableException {
-		Collection<Field> fields = cache.get(c);
+	public static Collection<Field> getFields(Class<?> type) throws NotSerializableException {
+		Collection<Field> fields = cache.get(type);
 		if (fields != null)
 			return fields;
 		fields = new ArrayList<>();
-		final Set<String> ids = new HashSet<>();
-		for (Class<?> sc = c; sc != null; sc = sc.getSuperclass()) {
-			final Field[] fs = sc.getDeclaredFields();
-			for (final Field f : fs) {
-				final int m = f.getModifiers();
-				if (Modifier.isStatic(m) || Modifier.isTransient(m))
+		Set<String> ids = new HashSet<>();
+		for (Class<?> superClass = type; superClass != null; superClass = superClass.getSuperclass()) {
+			Field[] declaredFields = superClass.getDeclaredFields();
+			for (Field field : declaredFields) {
+				int modifiers = field.getModifiers();
+				if (Modifier.isStatic(modifiers) || Modifier.isTransient(modifiers))
```

Comment:
This probably ought to ignore synthetic (created internally by the compiler) fields.